includes/Core/Util/Feature_Flags.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Feature_Flags
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use ArrayAccess;
+
+/**
+ * Class handling plugin activation.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Feature_Flags {
+	/**
+	 * Main instance for static access.
+	 *
+	 * @since n.e.x.t
+	 * @var Feature_Flags
+	 */
+	protected static $instance;
+
+	/**
+	 * Feature flag definitions.
+	 *
+	 * @since n.e.x.t
+	 * @var array|ArrayAccess
+	 */
+	private $features;
+
+	/**
+	 * Feature flag mode.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $mode;
+
+	/**
+	 * Gets the main feature flags instance.
+	 *
+	 * @since n.e.x.t
+	 * @internal
+	 *
+	 * @return Feature_Flags
+	 */
+	public static function get_instance() {
```

Comment:
I'm not sure an instance here is needed. While that's usually our preferred approach, we only need the static `enabled` method here. We should still be able to write integration tests via some internals static `set_config` or `set_mode` methods.

I'd prefer if we went truly static here which would also allow us to get rid of some unnecessary code.